src/routeSpec/RouteSpec.js

```diff
@@ -0,0 +1,52 @@
+'use strict'
+const QueryParameter = require('./QueryParameter')
+const _VALID_METHODS = ['CONNECT', 'DELETE', 'GET', 'HEAD', 'OPTIONS', 'PATCH', 'POST', 'PUT', 'TRACE']
+const _URI_SEGMENT_REGEX = '([a-zA-Z0-9\\$\\-_\\@\\.&\\+\\-\\!*"\'\\(\\)\\,](\\%[0-9]{2})?)+'
+const _PATH_PARAM_REGEX = `^${_URI_SEGMENT_REGEX}$`
+const _PATH_REGEX = `^\\/${_URI_SEGMENT_REGEX}$`
+const isValidPathDefinition = paths => paths.every(path => path.match(_PATH_REGEX))
+const isValidMethodDefinition = methods => methods.every(method => _VALID_METHODS.includes(method.toUpperCase()))
+const isValidPathParametersDefinition = pathParameters => pathParameters.every(param => param.match(_PATH_PARAM_REGEX))
+const isValidQueryParametersDefinition = queryParameters => queryParameters.every(param => param instanceof QueryParameter)
+const getPaths = paths => isValidPathDefinition(paths) ? paths : false
```

Comment:
I think getPaths should be paths=> allPathsAreValid(paths)? paths: throw new Error('...')